### DISCUSSION TITLE
Surface Shader independent from light

### DISCUSSION THREAD

**Discussion by Nucleon729** [marcofugaro/three-projected-material#69] [Ideas]
Hi

I'm reaching out because I'm exploring the possibility of customizing the shader used for the projection object in our project. Currently, I'm seeking to implement a "surface shader" that retains its appearance without reacting to changes in lighting conditions.
This would allow for a consistent visual presentation regardless of the surrounding light sources.

I would say rendered as 'flat shaded'

Would it be feasible to collaborate on this modification?

thank you

**Comment by Nucleon729** [comment]
@marcofugaro any idea?

**Reply by marcofugaro** [reply]
This isn't supported in the library, you would need to modify the code to extend [MeshBasicMaterial](https://threejs.org/docs/#api/en/materials/MeshBasicMaterial) instead of MeshPhysicalMaterial.
MeshBasicMaterial in three.js doesn't react to the lighting.

**Reply by Nucleon729** [reply]
That's great ! Will do that !
Thanks for the quick answer !
